tests/test_DM-12117.py

```diff
@@ -0,0 +1,106 @@
+# -*- coding: UTF-8 -*-
+
+#
+# LSST Data Management System
+# Copyright 2016 LSST Corporation.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+#
+
+import unittest
+import lsst.daf.persistence as dp
+import lsst.utils.tests
+import os
+import shutil
+import tempfile
+
+ROOT = os.path.abspath(os.path.dirname(__file__))
+
+
+def setup_module(module):
+    lsst.utils.tests.init()
+
+
+class MyTestMapper(dp.Mapper):
+
+    def __init__(self, root, *args, **kwargs):
+        self.root = root
+        self.args = args
+        self.kwargs = kwargs
+
+
+class TestDM12117(unittest.TestCase):
+    """Test case for basic Butler operations."""
+
+    def setUp(self):
+        self.testDir = tempfile.mkdtemp(dir=ROOT, prefix='test_DM-12117-')
+
+    def tearDown(self):
+        if os.path.exists(self.testDir):
+            shutil.rmtree(self.testDir)
+
+    @staticmethod
+    def repoBYaml(mapperArgs):
+        return """!RepositoryCfg_v1
+_mapper: 'lsst.daf.persistence.test.EmptyTestMapper'
+_mapperArgs: {}
+_parents: ['../repoA']
+_policy: null
+_root: null
+dirty: true
+""".format(mapperArgs)
+
+    def test(self):
+        """Test that an Old Butler parent repo that is can be loaded by a New Butler output repo and that the
+        output repo's mapper args are used by the OldButler repo.
+
+        1. create an Old Butler repo
+        2. create a New Butler repo without mapper args
+        3. reload that New Butler repo without naming its parent as an input
+        4. verify that the parent is loaded as an input
+
+        5. do the same, but specify a mapper arg, and verify that that mapper arg is passed to the parent as
+           well as the root repo.
+        """
+        for mapperArgs in ({}, {'calib': 'foo'}):
```

Comment:
Doing this as a for loop makes it hard to parse what's going on. I'd suggest writing a helper method `_testSomething/checkSomething` that takes mapperArgs and does what is inside the loop, and then unrolling the loop like this (if I'm interpreting it correctly):

```python
def testSomething(self):
_testSomething({})
_testSomething({'calib':'foo'})
```

This will make test failures much easier to debug, since you won't have to figure out which loop is failing.

Actually, since you're calling `tearDown` manually at the bottom of this, I'd recommend just making each of the above calls its own test method: `testSomethingDefaults` and `testSomethingCalib`. Then `tearDown` will behave as expected. And the method docstring here can go up to the class level.